### DISCUSSION TITLE
Is the `None` value of the scalar even possible for the `exists`?

### DISCUSSION THREAD

**Discussion by sz3lbi** [sqlalchemy/sqlalchemy#9733] [Usage Questions] [answered]
Hi there,
I have the following code to check if a person is a teacher:
```python
def is_teacher(self) -> bool:
    with Session() as session:
        statement = exists().where(TeacherOrm.person_id == self._id).select()
        teacher_orm_exists = session.scalar(statement) # Any | None

        if teacher_orm_exists is None:
            return False

        return teacher_orm_exists
```

Without checking that `scalar` isn't `None`, Pylance will show me errors that types don't match. I use Pyright for static type checking.
My question is: is it even possible for this query to return anything other than `True` or `False`?

Thanks.

**Comment by CaselIT** [comment, marked as answer]
Hi,

That query no. But you could stick `.limit(0)` there and that would return `None`.

There are other methods that are typed not to return none, like https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.scalar_one

**Reply by sz3lbi** [reply]
I will use `scalar_one` in this case. Thanks for the quick and comprehensive answer.
